src/monitor-plan-workspace/monitor-plan.service.ts

```diff
@@ -1122,6 +1124,14 @@ export class MonitorPlanWorkspaceService {
             trx,
           )),
         ]),
+      ).filter(plan =>
+        plan.items.some(item =>
+          targetLocations.some(loc =>
+            loc.stackPipeId
+              ? item.unitId === loc.unitId && item['stackPipeId'] === loc.stackPipeId
+              : item.unitId === loc.unitId
```

Comment:
I don't think this will work for `items` that are unit stack configurations. A unit stack configuration will _always_ have both a unit ID and a stack pipe ID, while a monitor location (`loc`) will only have one or the other. Since we're just looking for one match, I'd suggest the following:
```suggestion
            this.isUnitDTO(item)
              ? item.unitId === loc.unitId
              : item.unitId === loc.unitId ||
                item.stackPipeId === loc.stackPipeId,
```